test/controllers/surveys_controller_test.rb

```diff
@@ -53,17 +64,20 @@ class SurveysControllerTest < ActionDispatch::IntegrationTest
   end
 
   test "should show survey" do
+    sign_in @user
     get survey_url(@survey)
     assert_response :success
   end
 
   test "should get edit" do
+    sign_in @user
     get edit_survey_url(@survey)
     assert_response :success
   end
 
   test "should update survey" do
-    patch survey_url(@survey), params: { survey: { name: @survey.name, published: @survey.published, user_id: @survey.user_id } }
+    sign_in @user
+    patch survey_url(@survey), params: { survey: { name: "Test", published: true, user_id: @survey.user_id } }
     assert_redirected_to survey_url(@survey)
```

Comment:
You should assert that the survey name has actually changed

Reply:
Thanks @spencerldixon this is a good point, I am working on that.